packages/core/src/types-hoist/envelope.ts

```diff
@@ -17,7 +17,7 @@ import type { SpanJSON } from './span';
 // Based on https://github.com/getsentry/relay/blob/b23b8d3b2360a54aaa4d19ecae0231201f31df5e/relay-sampling/src/lib.rs#L685-L707
 export type DynamicSamplingContext = {
   trace_id: string;
-  public_key: DsnComponents['publicKey'];
+  public_key?: DsnComponents['publicKey'];
```

Comment:
m: public_key is strictly required as per [develop spec](https://develop.sentry.dev/sdk/telemetry/traces/dynamic-sampling-context/#dsc-specification). I'd prefer having this reflected in our types. Am I missing something why we need to make it optional?

Reply:
this is already optional, as `DsnComponents['publicKey']` --> `string | undefined`. The only difference is requiring `undefined` to be passed or not 🤔